src/DebugEngineHost.Stub/DebugEngineHost.ref.cs

```diff
@@ -348,6 +348,16 @@ public static IDebugCodeContext2 GetDebugCodeContextForIntPtr(IntPtr contextId)
             throw new NotImplementedException();
         }
 
+
+        /// <summary>
+        /// In VS, calls Marshal.ReleaseCodeContextId on IDebugCodeContext2 to derefcount when we are done with object.
+        /// </summary>
+        /// <param name="codeContextId">In VS, the IDebugCodeContext2</param>
+        public static void ReleaseCodeContextId(IntPtr codeContextId)
```

Comment:
>ReleaseCodeContextId [](http://example.com/codeflow?start=27&length=20)

You didn't implement this in src\DebugEngineHost\HostMarshal.cs

Reply:
Actually, instead I would suggest undoing your changes to DebugEngineHost.ref.cs+AD7PendingBreakpoint.cs, and instead making this call from OpenDebugAD7. 

Otherwise: 
1. We are depending on something to always delete all pending breakpoints
-and-
2. All engines to use this opportunity to call ReleaseCodeContextId

It seems better to instead do this inside of OpenDebugAD7 when we create/destroy instruction breakpoints.